commit c781fbb25a8b820a86c513a104c04ca63f5c7dcc
Author: Jaegeuk Kim <jaegeuk@kernel.org>
Date:   Mon Feb 27 11:57:11 2017 -0800

    f2fs: avoid to issue redundant discard commands
    
    commit 8b107f5b97772c7c0c218302e9a4d15b4edf50b4 upstream.
    
    If segs_per_sec is over 1 like under SMR, previously f2fs issues discard
    commands redundantly on the same section, since we didn't move end position
    for the previous discard command.
    
    E.g.,
    
                           start  end
                             |    |
          prefree_bitmap = [01111100111100]
    
    And, after issue discard for this section,
                                 end      start
                                  |        |
          prefree_bitmap = [01111100111100]
    
    Select this section again by searching from (end + 1),
                                 start  end
                                    |   |
          prefree_bitmap = [01111100111100]
    
    Fixes: 36abef4e796d38 ("f2fs: introduce mode=lfs mount option")
    Cc: Damien Le Moal <damien.lemoal@wdc.com>
    Signed-off-by: Jaegeuk Kim <jaegeuk@kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1341_c781fbb25a8b820a86c513a104c04ca63f5c7dcc.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1341_c781fbb25a8b820a86c513a104c04ca63f5c7dcc.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
